### PR TITLE
Merging upstream into Manadalay repo

### PR DESCRIPTION
**Pull request by mukhoakash**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml [hunk 1]
```diff
@@ -94,6 +94,7 @@ spec:
             - "--nodeid=$(KUBE_NODE_NAME)"
             - "--metrics-address=0.0.0.0:{{ .Values.node.metricsPort }}"
             - "--drivername={{ .Values.driver.name }}"
+            - "--volume-attach-limit={{ .Values.driver.volumeAttachLimit }}"
```
Comment (nearora-msft) [inline charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml:97]: Shouldn't we apply these charts changes to charts/v2.0.0-alpha.1 charts too?

File: pkg/azuredisk/controllerserver.go [hunk 2]
```diff
@@ -186,7 +186,8 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		case fsTypeField:
 			// no op, only used in NodeStageVolume
 		case kindField:
-			// no op, only for compatibility
+			// fix csi migration issue: https://github.com/kubernetes/kubernetes/issues/103433
+			parameters[kindField] = string(v1.AzureManagedDisk)
```
Comment (nearora-msft) [inline pkg/azuredisk/controllerserver.go:190]: This change should to be copied to  pkg/provisioner/cloudprovisioner.go

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml (hunk 1 above)
Comment (mukhoakash) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml:97]: Yes. I missed it out for the v2 driver! I have added it.

File: pkg/azuredisk/controllerserver.go (hunk 2 above)
Comment (mukhoakash) [reply pkg/azuredisk/controllerserver.go:190]: Ah! I modified the cloudprovisioner but didnt `git add` this change here. Thanks for pointing out. Fixed

**Review by nearora-msft — APPROVED** [review]

**Review by sunpa93 — APPROVED** [review]

**Pull request merged**